django/contrib/auth/forms.py

```diff
@@ -155,7 +155,10 @@ class UserCreationForm(BaseUserCreationForm):
     def clean_username(self):
         """Reject usernames that differ only in case."""
         username = self.cleaned_data.get("username")
-        if username and User.objects.filter(username__iexact=username).exists():
+        if (
+            username
+            and self._meta.model.objects.filter(username__iexact=username).exists()
+        ):
```

Comment:
We should reuse the `unique` error message defined for fields and respect the overriding error messages in `Meta`:
```diff
diff --git a/django/contrib/auth/forms.py b/django/contrib/auth/forms.py
index 080e06f059..eaae0bf695 100644
--- a/django/contrib/auth/forms.py
+++ b/django/contrib/auth/forms.py
@@ -147,11 +147,6 @@ class BaseUserCreationForm(forms.ModelForm):
 
 
 class UserCreationForm(BaseUserCreationForm):
-    error_messages = {
-        **BaseUserCreationForm.error_messages,
-        "unique": _("A user with that username already exists."),
-    }
-
     def clean_username(self):
         """Reject usernames that differ only in case."""
         username = self.cleaned_data.get("username")
@@ -159,7 +154,15 @@ class UserCreationForm(BaseUserCreationForm):
             username
             and self._meta.model.objects.filter(username__iexact=username).exists()
         ):
-            raise forms.ValidationError(self.error_messages["unique"], code="unique")
+            self._update_errors(
+                ValidationError(
+                    {
+                        "username": self.instance.unique_error_message(
+                            self._meta.model, ["username"]
+                        )
+                    }
+                )
+            )
         else:
             return username
 
diff --git a/tests/auth_tests/test_forms.py b/tests/auth_tests/test_forms.py
index c548d4f65f..7463804bdf 100644
--- a/tests/auth_tests/test_forms.py
+++ b/tests/auth_tests/test_forms.py
@@ -378,6 +378,9 @@ class UserCreationFormTest(TestDataMixin, TestCase):
             class Meta(UserCreationForm.Meta):
                 model = ExtensionUser
                 fields = UserCreationForm.Meta.fields + ("date_of_birth",)
+                error_messages = {
+                    "username": {"unique": "This username has already been taken."}
+                }
 
         ExtensionUser.objects.create_user(
             username="testclient",
@@ -397,7 +400,7 @@ class UserCreationFormTest(TestDataMixin, TestCase):
         self.assertIs(form.is_valid(), False)
         self.assertEqual(
             form["username"].errors,
-            ["A user with that username already exists."],
+            ["This username has already been taken."],
         )
 
 
```